userspace/plugin/plugin_loader.c

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#ifdef _WIN32
+    typedef HINSTANCE library_handle_t;
+#else
+    #include <dlfcn.h>
+    typedef void* library_handle_t;
+#endif
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "plugin_loader.h"
+
+static void add_str_prefix(char* s, const char* prefix)
+{
+    char tmp[PLUGIN_MAX_ERRLEN];
+    strncpy(tmp, prefix, PLUGIN_MAX_ERRLEN - 1);
+    strncat(tmp, s, PLUGIN_MAX_ERRLEN - 1);
+    strcpy(s, tmp);
+}
+
+static void* getsym(library_handle_t handle, const char* name)
+{
+#ifdef _WIN32
+	return (void*) GetProcAddress(handle, name);
+#else
+	return (void*) dlsym(handle, name);
+#endif
+}
+
+// little hack for simplifying the plugin_load function
+#define SYM_RESOLVE(h, s) *(void **)(&(h->api.s)) = getsym(h->handle, "plugin_"#s);
+
+plugin_handle_t* plugin_load(const char* path, char* err)
+{
+    // alloc and init memory
+    plugin_handle_t* ret = (plugin_handle_t*) malloc (sizeof(plugin_handle_t));
+    memset(ret, 0, sizeof(plugin_handle_t));
```

Comment:
Any reason to not use `calloc` directly here?

Reply:
This is a good one! Thank you Mauro!